src/core/agent/services/signifyNotificationService.ts

```diff
@@ -166,7 +166,44 @@ class SignifyNotificationService extends AgentService {
     if (notif.r) {
       return;
     }
-
+    if (notif.a.r === NotificationRoute.ExnIpexGrant) {
+      const exchange = await this.props.signifyClient
+        .exchanges()
+        .get(notif.a.d);
+      const existingCredential = await this.props.signifyClient
+        .credentials()
+        .get(exchange.exn.e.acdc.d)
+        .catch(() => undefined);
+      const ourIdentifier = await this.identifierStorage
```

Comment:
I think this check can go inside the `if (existingCredential)` block

Reply:
@iFergal I put it here for this criteria "If we receive an IPEX grant for a credential that doesn’t exist, add a check to make sure we are the issuee before creating the notification"

Reply:
good point apologies